pkg/resourceapply/helpers.go

```diff
@@ -315,8 +315,17 @@ func ApplyGenericWithHandlers[T kubeinterfaces.ObjectInterface](
 	}
 
 	var recreateReason string
+	propagationPolicy := metav1.DeletePropagationBackground
 	if getRecreateReasonFunc != nil {
-		recreateReason = getRecreateReasonFunc(requiredCopy, existing)
+		var overridePropagationPolicy *metav1.DeletionPropagation
+		recreateReason, overridePropagationPolicy, err = getRecreateReasonFunc(requiredCopy, existing)
+		if err != nil {
+			return *new(T), false, fmt.Errorf("can't get recreate reason: %w", err)
+		}
+
+		if overridePropagationPolicy != nil {
+			propagationPolicy = *overridePropagationPolicy
+		}
```

Comment:
```suggestion
		recreateReason, propagationPolicy, err = getRecreateReasonFunc(requiredCopy, existing)
		if err != nil {
			return *new(T), false, fmt.Errorf("can't get recreate reason: %w", err)
		}
```

Reply:
(gh didn't allow me to suggest a change over non-changed+deleted line so this is in 3 changes but the essentially jsut eliminate the need for the override variable)

Reply:
done